client/src/modules/stock/entry/modals/lots.modal.js

```diff
@@ -2,21 +2,25 @@ angular.module('bhima.controllers')
   .controller('StockDefineLotsModalController', StockDefineLotsModalController);
 
 StockDefineLotsModalController.$inject = [
-  'appcache', '$uibModalInstance', 'uiGridConstants', 'data', 'LotService',
-  'SessionService', 'CurrencyService', 'NotifyService', 'StockEntryModalForm',
-  'bhConstants', '$translate', 'focus',
+  'appcache', '$uibModalInstance', 'uiGridConstants', 'data', 'SessionService',
+  'CurrencyService', 'NotifyService', 'bhConstants', 'StockEntryModalForm',
+  '$translate', 'focus', 'ExchangeRateService', 'LotService',
 ];
 
 function StockDefineLotsModalController(
-  AppCache, Instance, uiGridConstants, Data, Lots,
-  Session, Currencies, Notify, EntryForm,
-  bhConstants, $translate, Focus,
+  AppCache, Instance, uiGridConstants, Data, Session,
+  Currencies, Notify, bhConstants, EntryForm,
+  $translate, Focus, ExchangeRate, Lots,
 ) {
   const vm = this;
 
   const cache = new AppCache('StockEntryModal');
 
   // initialize the form instance
+  if (Data.stockLine.wac) {
+    Data.stockLine.unit_cost = Data.stockLine.wac;
```

Comment:
I suggest we leave this for @jniles to review.  I think that we should be saving the unit_cost for reference purposes.  So I think we should add a new lot field for the WAC and not overriding the old unit_cost.